sound/soc/codecs/max98373.c

```diff
@@ -596,6 +596,24 @@ static SOC_ENUM_SINGLE_DECL(max98373_adc_samplerate_enum,
 			    MAX98373_R2051_MEAS_ADC_SAMPLING_RATE, 0,
 			    max98373_ADC_samplerate_text);
 
+static int max98373_adc_get(struct snd_kcontrol *kcontrol,
+			    struct snd_ctl_elem_value *ucontrol)
+{
+	struct snd_soc_component *component = snd_kcontrol_chip(kcontrol);
+	struct max98373_priv *max98373 =
+		snd_soc_component_get_drvdata(component);
+	struct soc_mixer_control *mc =
+		(struct soc_mixer_control *)kcontrol->private_value;
+	int reg = 0;
+
+	if (snd_soc_dapm_get_bias_level(&component->dapm) != SND_SOC_BIAS_OFF)
+		regmap_read(max98373->regmap, mc->reg, &reg);
```

Comment:
@RanderWang is there any other codec driver which does this sort of things? If not we are going to have difficulties upstreaming this without explanations of why this is needed.

Reply:
@RanderWang I'm not sure if it is really necessary doing pm resume/suspend for the volatile register access while amp is suspended. Can we try to add 'pm_runtime_get_sync' before register read and then call 'pm_runtime_put_autosuspend' ?
Doing this in 'get' function kcontrol is also not common, but I don't have better idea here.

Reply:
@ryans-lee When will these kcontrol be used? IMO we don't need to know these value when amp is suspended, right? Maybe we don't need the ``ASoC: codec: max98373: Add get support for ADC widgets`` commit and just let it return error in suspended. In other words, I don't think it is a bug if the kcontrol return error when we try to get a value which is meaningless at that moment. What do you think?

Reply:
This control is being used to get PVDD level and chip temperature when calibration application is running. The calibration is performed mostly in the factory, but it could be called during amp run-time by calibration app.
Yes, I thought this is not a bug because we can't expect normal value while amp is suspended. @RanderWang What was the impact of -EBUSY error? Was it just because of access to 'MAX98373_R203E_AMP_PATH_GAIN'? not other volatile registers?